src/MLJInterface.jl

```diff
@@ -389,17 +393,284 @@ MLJModelInterface.metadata_model(
+    - regression: L2 loss or mse.
+    - regression_l1: L1 loss or mae.
+    - huber: Huber loss.
+    - fair: Fair loss.
+    - poisson: poisson regression.
+    - quantile: Quantile regression.
+    - mape: MAPE (mean mean_absolute_percentage_error) loss.
+    - gamma: Gamma regression with log-link.
+    - tweedie: Tweedie regression with log-link.
+- `categorical_feature::Vector{Int} = Vector{Int}()`: Used to specify the categorical features. Items in the vector are column indices representing which features should be interpreted as categorical.
+- `data_random_seed::Int = 1`: Random seed used when constructing histogram bins.
+- `is_sparse::Bool = true`: Enable/disable sparse optimization.
+- `is_unbalance::Bool = false`: Set to true if training data is unbalanced.
+- `boost_from_average::Bool = true`: Adjusts the initial score to the mean of labels for faster convergence.
+- `use_missing::Bool = true`: Whether or not to handle missing values.
+- `feature_pre_filter::Bool = true`: Whether or not to ignore unsplittable features.
+- `alpha::Float64 = 0.9`: Parameter used for huber and quantile regression.
+- `metric::Vector{String} = ["l2"]`: Metric(s) to be used when evaluating on evaluation set. For detailed information, see [the official documentation](https://lightgbm.readthedocs.io/en/latest/Parameters.html#metric-parameters)
+- `metric_freq::Int = 1`: The frequency to run metric evaluation at.
+- `is_training_metric::Bool = false`: Set true to output metric result on training dataset.
+- `ndcg_at::Vector{Int} = Vector{Int}([1, 2, 3, 4, 5])`: Evaluation positions for ndcg and map metrics.
+- `num_machines::Int = 1`: Number of machines to use when doing distributed learning.
+- `num_threads::Int  = 0`: Number of threads to use.
+- `local_listen_port::Int = 12400`: TCP listen port.
+- `time_out::Int = 120`: Socket timeout.
+- `machine_list_file::String = ""`: Path of files that lists the machines used for distributed learning.
+- `save_binary::Bool = false`: Whether or not to save the dataset to a binary file
+- `device_type::String = "cpu"`: The type of device being used. One of `cpu` or `gpu`
+- `force_col_wise::Bool = false`: Force column wise histogram building. Only applicable on cpu.
+- `force_row_wise::Bool = false`: Force row wise histogram building. Only applicable on cpu.
+- `truncate_booster::Bool = true`: Whether or not to truncate the booster.
+
+# Examples
+
+```julia
+
+using MLJ
+using PrettyPrinting
+import DataFrames
+import Statistics
+
+
+LGBMRegressor = @load LGBMRegressor
+
+features, targets = @load_boston
+features = DataFrames.DataFrame(features)
+@show size(features)
+@show targets[1:3]
+train, test = partition(collect(eachindex(targets)), 0.70, shuffle=true)
+
+first(features, 3) |> pretty
+lgb = LGBMRegressor() #initialised a model with default params
+lgbm = machine(lgb, features[train, :], targets[train, 1]) |> MLJ.fit!
+
+predict(lgbm, train)
```

Comment:
I think you mean `test` here. And as you are specifying row indices, not new observations, you need `predict(lgbm, rows=test)`, or better:

```suggestion
predict(lgbm, X[test, :])
```